### PR TITLE
[Beyond EVM] performant StateDB implementation - part 4

### PR DESCRIPTION
**Pull request by ramtinms**
see #5129

### CODE REVIEW

File: fvm/evm/stdlib/contract.go [hunk 1]
```diff
@@ -242,12 +159,19 @@ func newInternalEVMTypeEncodeABIFunction(
...
 			if err != nil {
 				panic(abiEncodingError{})
 			}
 
-			return interpreter.ByteSliceToByteArrayValue(inter, encodedValues)
+			encodedValues := interpreter.ByteSliceToByteArrayValue(inter, hexData)
+
+			invocation.Interpreter.ReportComputation(
+				environment.ComputationKindEVMEncodeABI,
+				uint(encodedValues.Count()),
```
Comment (janezpodhostnik) [inline fvm/evm/stdlib/contract.go:171]: nice change.

**Review by janezpodhostnik — APPROVED** [review]

File: fvm/evm/stdlib/contract.go (hunk 1 above)
Comment (m-Peter) [reply fvm/evm/stdlib/contract.go:171]: I guess this must have snuck back in by some conflict when merging in the master branch. This was just a naive implementation that I made as a first take :sweat_smile:
What we actually want in production, is the implementation that is currently on master.

File: fvm/evm/stdlib/contract.go [hunk 2]
```diff
@@ -100,77 +97,6 @@ func (e abiDecodingError) Error() string {
...
-				Type: element.StaticType(inter),
-			})
-		}
-
-		// continue iteration
-		return true
-	})
-}
-
 // EVM.encodeABI
```
Comment (sideninja) [inline fvm/evm/stdlib/contract.go:100]: remove?

File: fvm/evm/stdlib/contract.go [hunk 3]
```diff
@@ -209,15 +135,6 @@ func newInternalEVMTypeEncodeABIFunction(
 				panic(errors.NewUnreachableError())
 			}
```
Comment (sideninja) [inline fvm/evm/stdlib/contract.go:137]: how come this is removed now?

**Review by sideninja — APPROVED** [review]

File: fvm/evm/stdlib/contract.go (hunk 1 above)
Comment (ramtinms) [reply fvm/evm/stdlib/contract.go]: resynced master, there shouldn't be any changes to stdlib package now.

File: fvm/evm/stdlib/contract.go (hunk 3 above)
Comment (ramtinms) [reply fvm/evm/stdlib/contract.go]: fixed it, it was somehow not synced with the latest version of master.

**Review by m-Peter — APPROVED** [review]
LGTM!

**Pull request merged**